src/daos_vol_file.c

```diff
@@ -1854,6 +1866,166 @@ H5_daos_get_cont_open_task(H5_daos_cont_op_info_t *cont_op_info,
     D_FUNC_LEAVE;
 } /* end H5_daos_get_cont_open_task() */
 
+
+/*-------------------------------------------------------------------------
+ * Function:    H5_daos_get_cont_query_task
+ *
+ * Purpose:     Creates an asynchronous DAOS task that simply calls
+ *              daos_cont_query.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5_daos_get_cont_query_task(tse_task_cb_t prep_cb, tse_task_cb_t comp_cb, H5_daos_req_t *req,
+    tse_task_t **first_task, tse_task_t **dep_task)
```

Comment:
Just a minor point - you might consider using H5_daos_create_daos_task for this to reduce code needed as well as using a task off the task list to eliminate an allocation.

Reply:
sorry not sure how to do that when i need to create a DAOS task (with daos OPC).
the function you pointed me at seems to be used to call MPI or other sync functions.

Reply:
Are you sure you were looking at the right function? H5_daos_create_daos_task takes a DAOS OPC, num_deps/dep_tasks combo, a prep/comp cb and a private udata and then just performs everything below except for the task scheduling and incrementing of the req. rc/assignment of dep_task.

Reply:
ok i was looking at H5_daos_create_task instead since that was the one used in daos_vol_fie.
i see the usage of the other function including daos. let me switch to that then.

Reply:
done